server/src/test/java/org/elasticsearch/cluster/metadata/DataStreamMetadataTests.java

```diff
@@ -29,21 +32,17 @@ public void testFromXContent() throws IOException {
     @Override
     protected DataStreamMetadata createTestInstance() {
         if (randomBoolean()) {
-            return new DataStreamMetadata(ImmutableOpenMap.of(), ImmutableOpenMap.of());
-        }
-        ImmutableOpenMap.Builder<String, DataStream> dataStreams = ImmutableOpenMap.builder();
-        for (int i = 0; i < randomIntBetween(1, 5); i++) {
-            dataStreams.put(randomAlphaOfLength(5), DataStreamTestHelper.randomInstance());
+            return new DataStreamMetadata(Map.of(), Map.of());
```

Comment:
Since the internal representation of the DataStreamMetadata class is using ImmutableMap, we shouldn't be switching the test to use a JDK Map. This way we are extending the production code to fit the test.

Reply:
In the long-run, `DataStreamMetadata` will be modified to remove `ImmutableOpenMap` - this is a stop-gap to separate these two changes

Reply:
Indeed, this change should be done the other way round